mapgen/iwyu-mapgen-std-symbol.py

```diff
@@ -0,0 +1,564 @@
+from the LaTeX sources of the C++ standard.
+"""
+
+import argparse
+import glob
+import os
+import re
+import sys
+from itertools import repeat
+
+NAMESPACE_KWD = 'namespace'
+INLINE_KWD = 'inline'
+REQUIRES_KWD = 'requires'
+TEMPLATE_KWD = 'template'
+OPERATOR_KWD = 'operator'
+DECLTYPE_KWD = 'decltype'
+USING_KWD = 'using'
+DEFINE = 'define'
+TEXMACRO = re.compile(r'''@
+                      \\\w+         # Macro name with a preceding backslash.
+                      (\[[\w-]*\])? # Optional argument.
+                      (\{[\w-]*\})* # Mandatory arguments.
+                      @
+                      ''', re.VERBOSE)
+DEFNLIBXNAME = re.compile(r'@\\defnlibxname\{(\w*)\}')
+LIBGLOBAL = re.compile(r'@\\libglobal\{(\w*)\}')
+LIBMEMBER = re.compile(r'@\\libmember\{(\w*)\}')
+SYNOPSIS = re.compile(
+    r'''Header\ \\tcode\{<(?P<headername>\w+)>\}
+    \\?                     # An extra '\' is present in <iterator> synopsis.
+    \ synopsis
+    .*?                     # Search non-greedily.
+    \\begin\{codeblock(digitsep)?\}  # codeblockdigitsep is used for <ratio>.
+    (?P<code>.*?)
+    \\end\{codeblock''',
+    re.DOTALL | re.VERBOSE)
+
+OUTPUT_HEADER = \
+'C++ standard library mapping, produced with mapgen/iwyu-mapgen-std-symbol.py.'
+
+# Functions overloaded in different headers may cause problems because IWYU
+# doesn't consider function parameters in mappings. Hence they are excluded
+# for a while.
+OVERLOADED_FUNCS = {'std::abs', 'std::acos', 'std::acosh', 'std::asin',
+    'std::asinh', 'std::atan', 'std::atan2', 'std::atanh', 'std::begin',
+    'std::cos', 'std::cosh', 'std::div', 'std::end', 'std::erase',
+    'std::erase_if', 'std::exp', 'std::get', 'std::isalnum', 'std::isalpha',
+    'std::isblank', 'std::iscntrl', 'std::isdigit', 'std::isgraph',
+    'std::islower', 'std::isprint', 'std::ispunct', 'std::isspace',
+    'std::isupper', 'std::isxdigit', 'std::log', 'std::log10',
+    'std::make_error_code', 'std::make_error_condition', 'std::move',
+    'std::pow', 'std::print', 'std::println', 'std::remove', 'std::sin',
+    'std::sinh', 'std::sqrt', 'std::swap', 'std::tan', 'std::tanh',
+    'std::to_string', 'std::tolower', 'std::toupper', 'std::vprint_nonunicode',
+    'std::vprint_unicode'}
+
+# Some classes and template specializations are excluded as well.
+# The specializations defined in different headers differ only in C++20
+# constraints, and hence cannot be distinguished in the mapping. The classes are
+# provided by different headers, and some of the headers provide only forward-
+# declarations, therefore some additional consideration is required.
+DUPLICATED_CLASSES = {
+    'std::allocator',
+    'std::atomic',
+    'std::basic_common_reference<:0, :1, :2, :3>',
+    'std::compare_three_way',
+    'std::formatter<:0, :1>',
+    'std::hash',
+    'std::tm',
+    'std::tuple',
+}
+
+ERRONEOUS_SYMBOLS = dict(
+    (('std::formatter<std::chrono::hh_mm_ss<duration<:0, :1>>, :2>',
```

Comment:
Let's move the leading paren to the `dict(`, i.e.

```
ERRO...OLS = dict((
    ('std::formatter<std::chrono::hh_mm_ss<duration<:0, :1>>, :2>',
    'std::formatter<std::chrono::hh_mm_ss<std::chrono::duration<:0, :1>>, :2>'),
))
```

... and possibly align the second line by the quotes -- I think that's what Python editors prefer -- if you can live with overflowing the 80-col limit (I think it's fine in this case, since we don't have any control over the name length)